calculation/gmhazard_calc/gmhazard_calc/directivity/scripts/mc_nhyp_runs.py

```diff
@@ -24,20 +28,68 @@ def perform_mp_directivity(
     grid_space,
     nhm_dict,
     output_dir,
+    repeat_n_procs: int,
+    hypo_n_procs: int,
 ):
+    assert repeat_n_procs == 1 or hypo_n_procs == 1
+
     if nhypo is None:
         nhypo = hypo_along_strike * hypo_down_dip
     print(f"Computing for {fault_name} {nhypo}")
 
     fault, site_coords, planes, lon_lat_depth, x, y = directivity.utils.load_fault_info(
         fault_name, nhm_dict, grid_space
     )
-    n_hypo_data = directivity.NHypoData(method, nhypo, hypo_along_strike, hypo_down_dip)
+    n_hypo_data = directivity.NHypoData(
+        gmhazard_calc.HypoMethod(method), nhypo, hypo_along_strike, hypo_down_dip
+    )
 
-    total_fd = np.zeros((repeats, len(site_coords), 1))
-    total_fd_array = np.zeros((repeats, nhypo, len(site_coords), 1))
 
-    for i in range(repeats):
+    if n_hypo_data.method in [gmhazard_calc.HypoMethod.MONTE_CARLO, gmhazard_calc.HypoMethod.LATIN_HYPERCUBE]:
+        total_fd = np.zeros((repeats, len(site_coords), 1))
+        total_fd_array = np.zeros((repeats, nhypo, len(site_coords), 1))
+
+        if repeat_n_procs == 1:
+            for i in range(repeats):
+                fdi, fdi_array, phi_red = directivity.compute_fault_directivity(
```

Comment:
`phi_red` never used, could be an underscore?
(There are two `phi_red`, one from this single process and another one in multiprocessing)